internal/pkg/policy/standalone.go

```diff
@@ -87,17 +81,32 @@ func (m *standAloneSelfMonitorT) State() client.UnitState {
 	return m.state
 }
 
-func (m *standAloneSelfMonitorT) check(ctx context.Context) client.UnitState {
+func (m *standAloneSelfMonitorT) check(ctx context.Context) {
+	ctx, cancel := context.WithTimeout(ctx, m.checkTimeout)
+	defer cancel()
+
+	current := m.State()
+	state := client.UnitStateHealthy
+	message := "Running"
+
 	_, err := m.policyF(ctx, m.bulker, dl.WithIndexName(m.policiesIndex))
-	if err != nil {
-		if errors.Is(err, es.ErrIndexNotFound) {
-			m.log.Debug().Str("index", m.policiesIndex).Msg(es.ErrIndexNotFound.Error())
-			return m.updateState(client.UnitStateHealthy, "Running: Policies not available yet")
-		}
-		if err != nil {
-			return m.updateState(client.UnitStateFailed, fmt.Sprintf("Failed to request policies: %s", err))
+	if errors.Is(err, es.ErrIndexNotFound) {
+		m.log.Debug().Str("index", m.policiesIndex).Msg(es.ErrIndexNotFound.Error())
+		message = "Running: Policies not available yet"
```

Comment:
@michel-laterman should it be actually healthy if the policies index hasn't been created yet? Would requests succeed if Fleet Server cannot use this index? :thinking:

Reply:
IMO it should be unhealthy if the index does not exist, that way the fleet-controller/k8s can use it for health checks/traffic routing. What do you think @nchaulet?
This PR also changes the status endpoint to return 200 only on a healthy status

Reply:
> IMO it should be unhealthy if the index does not exist, that way the fleet-controller/k8s can use it for health checks/traffic routing. What do you think @nchaulet?

Not sure the index while not exists if the user do not create any agent policy, so it probably should be healthy even if the index do not exists

Reply:
Sounds good, let's add the behaviour to the description in the openapi doc

Reply:
Description updated in https://github.com/elastic/fleet-server/pull/2693/files#diff-5ad174233c27653aa08c5ce213ddfc13cbf7bcfc10cdf724a547da655e5743c0R830, let me know if this is fine.